File: app/src/main/java/studentDatabase/Pojo/Dept.java
package studentDatabase.Pojo;

import lombok.Data;
import lombok.NonNull;

@Data
public class Dept {
    @NonNull 
    private String deptid;
    private String deptname;
    private String deptaddress;
    private int deptphone;
    
    public Dept() {
    } // needed for JSON serialization
    
    public Dept(String deptid, String deptname, String deptaddress, int deptphone) {
        this.deptid = deptid;
        this.deptname = deptname;
        this.deptaddress = deptaddress;
        this.deptphone = deptphone;
    }
    
    public void setDeptid(String deptid) throws Exception {
        if (deptid.isBlank()) {
            throw new Exception("Empty bean id.");
        }
        this.deptid = deptid.toUpperCase();
    }
    
}

File: app/src/main/resources/public/forms/dept.js
// import {navcomp} from '/components/nav.js';

'use strict'

class dept {
    deptid;
    deptname;
    deptaddress;
    deptphone;
}

const schemabean = new dept();
const fields = Object.keys(schemabean);


const formApp = {
    data() {
        return {
            message: "Hello",
            bean: schemabean,
            fields: fields,
        }
    },
    computed: {},
    methods: {
        test() {
            console.log("hello world");
        },
        async submitBean() {
            const resp = await fetch("/api/dept/add", {
                method: 'PUT',
                cache: 'no-cache',
                body: JSON.stringify(bean)
            });

            const msg = await resp.text();
            alert(msg);
        }
    }
}

const app = Vue.createApp(formApp)
// navcomp(app);
const vm = app.mount("#deptform")
